src/Data/List/Relation/Suffix/Heterogeneous/Properties.agda

```diff
@@ -0,0 +1,233 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of the heterogeneous suffix relation
+------------------------------------------------------------------------
+
+module Data.List.Relation.Suffix.Heterogeneous.Properties where
+
+open import Function using (_$_; flip)
+open import Relation.Nullary using (Dec; yes; no)
+import Relation.Nullary.Decidable as Dec
+open import Relation.Unary as U using (Pred)
+open import Relation.Nullary.Negation using (contradiction)
+open import Relation.Binary as B using (REL; Rel; Trans; Antisym; Irrelevant; _⇒_)
+open import Relation.Binary.PropositionalEquality as P using (_≡_; refl; sym; subst)
+open import Data.Nat as N using (suc; _+_; _≤_; _<_)
+open import Data.List as List using (List; []; _∷_; _++_; length; filter; replicate; reverse)
+open import Data.List.Relation.Pointwise as Pw using (Pointwise; []; _∷_; Pointwise-length)
+open import Data.List.Relation.Suffix.Heterogeneous as Suffix using (Suffix; here; there; tail)
+open import Data.List.Relation.Prefix.Heterogeneous as Prefix using (Prefix)
+import Data.Nat.Properties as ℕₚ
+import Data.List.Properties as Listₚ
+import Data.List.Relation.Prefix.Heterogeneous.Properties as Prefixₚ
+
+------------------------------------------------------------------------
+-- reverse (convert to and from Prefix)
+
+module _ {a b r} {A : Set a} {B : Set b} {R : REL A B r} where
+
+  -- TODO: move this lemma (and others) into Pointwise?
+  private
+    pw-reverse : ∀ {as bs} → Pointwise R as bs → Pointwise R (reverse as) (reverse bs)
+    pw-reverse = foldl []
+      where
+      foldl : ∀ {as bs as′ bs′} → Pointwise R as′ bs′ → Pointwise R as bs →
+              Pointwise R (List.foldl (flip _∷_) as′ as) (List.foldl (flip _∷_) bs′ bs)
+      foldl rs′ []       = rs′
+      foldl rs′ (r ∷ rs) = foldl (r ∷ rs′) rs
+
+  fromPrefix⁺ : ∀ {as bs} → Prefix R as bs → Suffix R (reverse as) (reverse bs)
```

Comment:
This seems like `fromPrefix` to me (no need for the `⁺` as the conclusion
does not mention a function called `fromPrefix`).

It should probably go in `Suffix.Heterogeneous` proper rather than `X.Properties`.

Reply:
Oh I see. The problem is that `toPrefix⁻` is eerily similar. Hmm.

Reply:
Hmm I agree with @gallais that these names don't obey the conventions used elsewhere. I think better names for these lemmas might be `fromPrefix`, `fromPrefix-rev`, `toPrefix-rev` and `toPrefix` respectively...